pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -132,6 +147,102 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	return nil
 }
 
+// getAllDependencies gets all the dependencies for the purls provided in a concurrent manner.
+func (d *depsCollector) getAllDependencies(ctx context.Context, purls []datasource.Source) error {
+	// channels to signal when the project and version info have been fetched
+	projectDone := make(chan bool)
+	versionDone := make(chan bool)
+
+	// channels to send the inputs to the goroutines
+	projectChan := make(chan *pb.ProjectKey)
+	versionChan := make(chan *pb.VersionKey)
+	logger := logging.FromContext(ctx)
+
```

Comment:
it would be heflpul to add a comment to show the pipeline of the channels here at the top (and the go routines).

Reply:
I added a few comments. Do they seem acceptable to you?